### DISCUSSION TITLE
determine weights from orientations

### DISCUSSION THREAD

**Discussion by makeabhishek** [mtex-toolbox/mtex#1472] [Ask Anything]
Hi
I have an EBSD (.ang) file. I want to calculate the odf and determine the weights. How can I determine the weights? I tried to calculate odf density but it does not provide the weights, however, it generates Fourier components. if I generate pole figure data and generate odf from that it generates fewer weights for example only 100 weights for 500 grains.
`
ebsd = EBSD.load(fileName);
ori = ebsd(phase1).orientations;

odf = calcDensity(ori)

h=Miller({0,0,0,1},{1,0,-1,0},{2,-1,-1,0},{0,-1,1,0},cs);
r = regularS2Grid('resolution',5*degree);
pf = calcPoleFigure(odf,h,r)
odf = calcODF(pf)
`

**Comment by lfgmorales** [comment]
Can you elaborate what do you mean exactly by weights here?


cheers


Luiz

**Comment by makeabhishek** [comment]
I mean the texture weights. 

I think it can be export by vpsc. 

Best Regards
Abhishek

**Comment by lfgmorales** [comment]
It is still not clear (at least to me) what do you want to achieve, but maybe that helps

https://mtex-toolbox.github.io/ODFCharacteristics.html

Cheers

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 28 Jun 2022, at 14:25, Abi ***@***.***> wrote:

﻿

I mean the texture weights.

I think it can be export by vpsc.

Best Regards
Abhishek

> On 28 Jun 2022, at 8:12 PM, Luiz Morales ***@***.***> wrote:
>
> ﻿
> Can you elaborate what do you mean exactly by weights here?
>
>
> cheers
>
>
> Luiz
>
>
>
>
> > On 28 Jun 2022, at 13:37, Abi ***@***.***> wrote:
> >
> >
> > Hi
> > I have an EBSD (.ang) file. I want to calculate the odf and determine the weights. How can I determine the weights? I tried to calculate odf density but it does not provide the weights, however, it generates Fourier components. if I generate pole figure data and generate odf from that it generates fewer weights for example only 100 weights for 500 grains.
> > `
> > ebsd = EBSD.load(fileName);
> > ori = ebsd(phase1).orientations;
> >
> > odf = calcDensity(ori)
> >
> > h=Miller({0,0,0,1},{1,0,-1,0},{2,-1,-1,0},{0,-1,1,0},cs);
> > r = regularS2Grid('resolution',5*degree);
> > pf = calcPoleFigure(odf,h,r)
> > odf = calcODF(pf)
> > `
> >
> > —
> > Reply to this email directly, view it on GitHub <https://github.com/mtex-toolbox/mtex/discussions/1472>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ALE5INYCYK4Y47DRBAJUYBDVRLPXTANCNFSM52B3AJPQ>.
> > You are receiving this because you are subscribed to this thread.
> >
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1472#discussioncomment-3039687>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN23G7EV5G34TZJSAQLVRLVKDANCNFSM52B3AJPQ>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1472#discussioncomment-3039687", "url": "https://github.com/mtex-toolbox/mtex/discussions/1472#discussioncomment-3039687", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by makeabhishek** [comment]
Thanks, Luiz for sharing the link. What I meant is to calculate the of
weights for a particular orientation. If I want to generate a synthetic
texture, I need to have the orientation and corresponding weightages. So I
want to obtain the weights and orientation from a given .and file.

On Tue, Jun 28, 2022 at 10:50 PM Luiz Morales ***@***.***>
wrote:

> It is still not clear (at least to me) what do you want to achieve, but
> maybe that helps
>
> https://mtex-toolbox.github.io/ODFCharacteristics.html
>
> Cheers
>
> Luiz
>
> Luiz F. G. Morales
> ScopeM - ETH Zürich
> Sent from iPhone
>
> On 28 Jun 2022, at 14:25, Abi ***@***.***> wrote:
>
> ﻿
>
> I mean the texture weights.
>
> I think it can be export by vpsc.
>
> Best Regards
> Abhishek
>
> > On 28 Jun 2022, at 8:12 PM, Luiz Morales ***@***.***> wrote:
> >
> > ﻿
> > Can you elaborate what do you mean exactly by weights here?
> >
> >
> > cheers
> >
> >
> > Luiz
> >
> >
> >
> >
> > > On 28 Jun 2022, at 13:37, Abi ***@***.***> wrote:
> > >
> > >
> > > Hi
> > > I have an EBSD (.ang) file. I want to calculate the odf and determine
> the weights. How can I determine the weights? I tried to calculate odf
> density but it does not provide the weights, however, it generates Fourier
> components. if I generate pole figure data and generate odf from that it
> generates fewer weights for example only 100 weights for 500 grains.
> > > `
> > > ebsd = EBSD.load(fileName);
> > > ori = ebsd(phase1).orientations;
> > >
> > > odf = calcDensity(ori)
> > >
> > > h=Miller({0,0,0,1},{1,0,-1,0},{2,-1,-1,0},{0,-1,1,0},cs);
> > > r = regularS2Grid('resolution',5*degree);
> > > pf = calcPoleFigure(odf,h,r)
> > > odf = calcODF(pf)
> > > `
> > >
> > > —
> > > Reply to this email directly, view it on GitHub <
> https://github.com/mtex-toolbox/mtex/discussions/1472>, or unsubscribe <
> https://github.com/notifications/unsubscribe-auth/ALE5INYCYK4Y47DRBAJUYBDVRLPXTANCNFSM52B3AJPQ>.
>
> > > You are receiving this because you are subscribed to this thread.
> > >
> >
> > —
> > Reply to this email directly, view it on GitHub, or unsubscribe.
> > You are receiving this because you authored the thread.
>
> —
> Reply to this email directly, view it on GitHub<
> https://github.com/mtex-toolbox/mtex/discussions/1472#discussioncomment-3039687>,
> or unsubscribe<
> https://github.com/notifications/unsubscribe-auth/ALE5IN23G7EV5G34TZJSAQLVRLVKDANCNFSM52B3AJPQ>.
>
> You are receiving this because you commented.Message ID: ***@***.***>
>
> [ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage",
> "potentialAction": { ***@***.***": "ViewAction", "target": "
> https://github.com/mtex-toolbox/mtex/discussions/1472#discussioncomment-3039687",
> "url": "
> https://github.com/mtex-toolbox/mtex/discussions/1472#discussioncomment-3039687",
> "name": "View Discussion" }, "description": "View this Discussion on
> GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub",
> "url": "https://github.com" } } ]
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mtex-toolbox/mtex/discussions/1472#discussioncomment-3040897>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALNXRFEPFXRZ3TFPWAAEU7TVRMGJ3ANCNFSM52B3AJPQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks & Regards
Abhishek Saini


This email and any attached files are confidential and intended solely for
the intended recipient(s). If you are not the named recipient you should
not read, distribute copy or alter this email. Any views or opinions
expressed in this email are those of the author and do not represent those
of the organization. Warning: Although precautions have been taken to make
sure no viruses are present in this email, the organization cannot accept
responsibility for any loss or damage that arise from the use of this email
or attachments.

**Comment by zmichels** [comment]
Do you mean the M.U.D. / M.R.D. for any given orientation? If so, you may want to generate a grid and compute a kernel density at each point on the grid.

Zach

**Reply by kilir** [reply]
e.g.
```
cs = crystalSymmetry('23')
o = orientation.rand(100,cs)
odf = calcDensity(o)
% evaluate at o
w = eval(odf,o)

% or on a grid
g = equispacedSO3Grid(cs)
w = eval(odf,g)
```
